documentation/developer_setup.md

```diff
@@ -0,0 +1,336 @@
+	cred    SharedKeyCredential
+}
+
+```
+Note that there are no exported fields on the `ServiceClient` struct, and as a rule of thumb, generated clients and credentials should be private.
+
+Constructors for clients are separate methods that are not associated with the struct. The constructor for the ServiceClient is as follow:
+```golang
+// NewServiceClient creates a ServiceClient struct using the specified serviceURL, credential, and options.
+func NewServiceClient(serviceURL string, credential azcore.Credential, options *ClientOptions) (*ServiceClient, error) {
+	conOptions := options.getConnectionOptions()
+	if isCosmosEndpoint(serviceURL) {
+		conOptions.PerCallPolicies = []azcore.Policy{CosmosPatchTransformPolicy{}}
+	}
+	con := newConnection(serviceURL, cred, conOptions)
+	c, err := cred.(*SharedKeyCredential)
+	return &ServiceClient{client: &tableClient{con}, service: &serviceClient{con}, cred: *c}, err
+}
+```
+In `Go`, the method parameters are enclosed with parenthesis immediately following the method name with the parameter name preceding the parameter type. The return arguments follow the parameters. If a method has more than one return parameter the types of the parameter must be enclosed in parenthesis. Note the `*` before a type indicates a pointer to that type. All methods that create a new client or interact with the service should return an `error` type as the last argument.
+
+This client takes three parameters, the first is the service URL for the specific account. The second is an [`interface`][go_interfaces] which is a specific struct that has definitions for a certain set of methods. In the case of `azcore.Credential` the `AuthenticationPolicy(options AuthenticationPolicyOptions) Policy` method must be defined to be a valid interface. The final argument to methods that create clients or interact with the service should be a pointer to an `Options` parameter. Making this final parameter a pointer allows the customer to pass in `nil` if there are no specific options they want to change. The `Options` type should have a name that is intuitive to what the customer is trying to do, in this case `ClientOptions`.
+
+
+### Defining Methods
+
+Defining a method follows the format:
+```golang
+// Create creates a table with the specified name.
+func (s *ServiceClient) Create(ctx context.Context, name string) (TableResponseResponse, error) {
+	resp, err := t.client.Create(ctx, TableProperties{&name}, new(TableCreateOptions), new(QueryOptions))
+	if err == nil {
+		tableResp := resp.(TableResponseResponse)
+		return tableResp, nil
+	}
+	return TableResponseResponse{}, err
+}
+```
+The `(s *ServiceClient)` portion is the "receiver". Methods can be defined for either pointer (with a `*`) or receiver (without a `*`) types. Pointer receivers will not copy types on method calls and allows the method to mutate the receiving struct. It is best practice to use pointer receivers wherever possible to limit memory copies.
```

Comment:
The "best practice" is not accurate.  There is a lot of discussion about byval/byref receivers and method sets.  Without going too much into the weeds on this, I think you could simply say "Client methods should use a pointer receiver."